src/file.ts

```diff
@@ -2763,18 +2775,23 @@ class File extends ServiceObject<File, FileMetadata> {
       const policyBase64 = Buffer.from(policyString).toString('base64');
 
       try {
-        const signature = await this.storage.authClient.sign(policyBase64);
+        const signature = await this.storage.authClient.sign(
+          policyBase64,
+          options.signingEndpoint
+        );
         const signatureHex = Buffer.from(signature, 'base64').toString('hex');
+        const universe = this.parent.storage.universeDomain;
         fields['policy'] = policyBase64;
         fields['x-goog-signature'] = signatureHex;
 
         let url: string;
+
         if (options.virtualHostedStyle) {
-          url = `https://${this.bucket.name}.storage.googleapis.com/`;
+          url = `https://${this.bucket.name}.storage.${universe}/`;
```

Comment:
So I think this still doesn't address the custom endpoint on client use-case (ie. when an endpoint is set on the client, then any signURL created using that client will point to that endpoint instead of the default storage.googleapis.com).

However, that can be a separate FR as this works for TPC. I'm happy as long as the TPC portion of the recently added conformance tests pass, or there's a similar unit test which verifies that calling sign URL will produce a url pointing to the TPC universe (you can extract the signature from the conformance test to check that as well)

Reply:
> So I think this still doesn't address the custom endpoint on client use-case (ie. when an endpoint is set on the client, then any signURL created using that client will point to that endpoint instead of the default storage.googleapis.com).

Ah, for post policies too. I'll add that.